resolver.go

```diff
@@ -32,8 +32,11 @@ type resolver struct {
 }
 
 func (c *resolver) ResolveNow(grpcresolver.ResolveNowOptions) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
+	locked := c.mu.TryLock()
+	if !locked { // already resolving
+		return
+	}
```

Comment:
Don't we need to wait until the resolution is finished?
It is likely to reuse the old(problematic) version again.

Reply:
@KimMachineGun 
I referred grpc-go code, and `ResolveNow()` is called from `ccResolverWrapper` :
https://github.com/grpc/grpc-go/blob/v1.56.2/resolver_conn_wrapper.go#L116

And it only call from `ClientConn` :
https://github.com/grpc/grpc-go/blob/v1.56.2/clientconn.go#L1168

You can find this:
```go
go r.resolveNow(o)
```

As you know, it means it does not wait resolving.